### ISSUE TITLE
Check if Content is empty before hashing the request body.

### ISSUE THREAD

**Issue by AlexandreWambre** [softawaregmbh/library-authentication#19]
Hello team,

thank you for your work.

Sometime, content can be empty in SendAsync func (some library are doing this) and application crash. Can you add a security on it?

We have found that things are going well if you add ```if(content.Length > 0) ``` just after ```var content = await request.Content.ReadAsByteArrayAsync();``` and don't hash the body.

result will be :
```
var content = await request.Content.ReadAsByteArrayAsync();
if (content.Length > 0) {
        using (var hashAlgorithm = this.requestBodyHashingMethod.CreateHashAlgorithm())
        {
            // Hashing the request body, any change in request body will result in different hash, we'll incure message integrity
            var requestContentHash = hashAlgorithm.ComputeHash(content);
            requestContentBase64String = Convert.ToBase64String(requestContentHash);
        }
}
```

https://github.com/softawaregmbh/library-authentication/blob/9e726c01bbbb965ea75f632ee42eeed09d67f132/softaware.Authentication.Hmac.Client/ApiKeyDelegatingHandler.cs#L77

**Comment by meinsiedler** [comment]
Hi @AlexandreWambre ,

thanks for reporting the issue! We'll take a look at it and provide a new version.

**Comment by meinsiedler** [comment]
Hi @AlexandreWambre ,

we just released [version 2.0.1](https://www.nuget.org/packages/softaware.Authentication.Hmac.Client/2.0.1) of our HMAC client package which fixes this issue. 🎉

**Note:** If you're not on version 2.0.0 already, please be aware of breaking changes from version 1.2.0 to 2.0.0. Take a look at the [release notes](https://github.com/softawaregmbh/library-authentication/releases/tag/4.0.0) for details.

**Comment by AlexandreWambre** [comment]
Great! thank you!
